### PR TITLE
Add map type toggle to place_tracker demo.

### PR DESCRIPTION
**Pull request by kenzieschmoll**
Added a button that toggles between different map types: satellite, terrain, hybrid, and normal.

### CODE REVIEW

**Comment by kenzieschmoll** [comment]
screenshot_1541021563

File: place_tracker/lib/place_map.dart [hunk 1]
```diff
@@ -234,6 +234,21 @@ class PlaceMapState extends State<PlaceMap> {
...
   }
 
+  void _onToggleMapTypePressed() {
+    MapType nextType =
+      MapType.values[(mapController.options.mapType.index + 1) % MapType.values.length];
+
+    // We do not want to show MapType.none, so skip over it.
+    if (nextType == MapType.none) {
+      nextType =
+        MapType.values[(mapController.options.mapType.index + 2) % MapType.values.length];
```
[the remark this answers is not in the document]
Comment (kenzieschmoll) [reply place_tracker/lib/place_map.dart]: No, the +2 will cause MapType.none to be skipped. But okay, I'll just keep MapType.none in the cycle.

File: place_tracker/lib/place_map.dart [hunk 2]
```diff
@@ -278,9 +293,10 @@ class PlaceMapState extends State<PlaceMap> {
                 onSavePressed: () => _confirmAddPlace(context),
                 onCancelPressed: _cancelAddPlace,
               ),
-              _AddPlaceFab(
+              _MapFabs(
```
[the remark this answers is not in the document]
Comment (kenzieschmoll) [reply place_tracker/lib/place_map.dart]: This is possible, but then when the snackbar comes up saying "New placed added", it slides the fabs up with it, which looks weird. If I add the fabs to the stack, this is not a problem.

File: place_tracker/lib/place_map.dart [hunk 3]
```diff
@@ -234,6 +234,15 @@ class PlaceMapState extends State<PlaceMap> {
     }
   }
 
+  void _onToggleMapTypePressed() {
+    MapType nextType =
```
Comment (HansMuller) [inline place_tracker/lib/place_map.dart]: final

File: place_tracker/lib/place_map.dart [hunk 4]
```diff
@@ -400,33 +410,49 @@ class  _AddPlaceButtonBar extends StatelessWidget {
...
-          child: FloatingActionButton(
-            onPressed: onPressed,
-            materialTapTargetSize: MaterialTapTargetSize.padded,
-            backgroundColor: Colors.green,
-            child: const Icon(Icons.add_location, size: 36.0),
-          ),
+    return Container(
+      alignment: Alignment.topRight,
+      margin: const EdgeInsets.only(top: 12.0, right: 12.0),
+      child: Opacity(
```
Comment (HansMuller) [inline place_tracker/lib/place_map.dart]: The Visibility widget would be a little better here.

File: place_tracker/lib/place_map.dart [hunk 5]
```diff
@@ -400,33 +410,49 @@ class  _AddPlaceButtonBar extends StatelessWidget {
...
+        opacity: visible ? 1.0 : 0.0,
+        child: Column(
+          children: <Widget>[
+            FloatingActionButton(
+              heroTag: "add_place_btn",
```
Comment (HansMuller) [inline place_tracker/lib/place_map.dart]: We don't usually abbreivate. Typically strings are `'foo'` rather than `"foo"`.

**Pull request merged**